docs(running.md): update docs with kubernetes deployment workflow

Anyitechs opened the pull request:
This PR documents the Kubernetes deployment workflow/commands in `/justfile` into `running.md`. It follows the first option [here](https://github.com/bitcoin-dev-project/warnet/issues/308#issuecomment-2042590258) to document the commands explicitly, and closes issue #308

Commit:
docs(running.md): update docs with kubernetes deployment workflow

pinheadmz commented on docs/running.md:
```diff
@@ -9,12 +9,110 @@ a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet
+kubectl apply -f src/templates/rpc/warnet-rpc-service.yaml
+
+# Deploys a statefulset for the RPC to manage stateful applications
+kubectl apply -f src/templates/rpc/warnet-rpc-statefulset-dev.yaml
```
the `-dev.yaml` is for local development (changes in the warnet codebase are reflected in the container) so we can probably just chop that off and illustrate use of the production image which is just `warnet-rpc-statefulset.yaml`. That applies to this entire document.

Anyitechs replied:
Ok. I'll update that.

pinheadmz reviewed (commented):
Really good thank you! Besides the -dev thing mentioned below, I think we should also 

1) Mention the just command that macro's each of these. So at the top you can mention the `just` option, and then for each section just add the corresponding (example) `just startd`

2) I think we should also include port forwarding which is required to send RPC commands from local user into the cluster. That is started with `just p` from the justfile and the fill command is like `kubectl portforward 9276....` something something something.


Thank you!

Anyitechs commented:
> Really good thank you! Besides the -dev thing mentioned below, I think we should also
> 
> 1. Mention the just command that macro's each of these. So at the top you can mention the `just` option, and then for each section just add the corresponding (example) `just startd`
> 2. I think we should also include port forwarding which is required to send RPC commands from local user into the cluster. That is started with `just p` from the justfile and the fill command is like `kubectl portforward 9276....` something something something.
> 
> Thank you!

Thank you for the review. I'll work on these and update the PR.

mplsgrant reviewed (commented):
As a new user, I want to run most of these commands using the `justfile`. Is it possible to direct users to the `justfile` features and then give them the option to explore the nitty gritty details? 

One idea is to provide the justfile command up front and then give users the option to explore that command in more detail using [collapsing sections](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab)

Running Warnet
------


`just start` - starts a minikube cluster
<details>
  <summary>start details</summary>

  ```bash
# Mounts the local source directory to the Minikube virtual machine, allowing the RPC service to access code and files from the local development environment.
minikube mount $PWD:/mnt/src > /tmp/minikube_mount.log 2>&1 &

# Capture the PID of the minikube mount command
MINIKUBE_MOUNT_PID=$!

# Save the PID to a file for later use
echo $MINIKUBE_MOUNT_PID > /tmp/minikube_mount.pid

# Setup k8s
kubectl apply -f src/templates/rpc/namespace.yaml
kubectl apply -f src/templates/rpc/rbac-config.yaml
kubectl apply -f src/templates/rpc/warnet-rpc-service.yaml
kubectl apply -f src/templates/rpc/warnet-rpc-statefulset-dev.yaml
kubectl config set-context --current --namespace=warnet

# Wait for the RPC pod to become ready, ensuring that the service is fully operational before proceeding.
sleep 2
kubectl wait --for=condition=Ready --timeout=2m pod rpc-0
  ```
</details>


`just startd` - starts a minikube cluster in debug mode
<details>
  <summary>startd details</summary>

  ```bash
Thing 1
Thing 2
  ```
</details>

mplsgrant commented:
When we run the network using minikube, is it still a "Docker network"? I noticed that the documentation talks about minikube and docker as two options to run the system, however, when I eventually run `warcli network start`, it gives me a result that talks about which Docker network I am using:

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Temp dir                              ┃ Bitcoin network ┃ Docker network ┃ Subnet      ┃ Graph                                 ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ /home/grant/.warnet/warnet/…          │ regtest         │ warnet         │ 100.0.0.0/8 │ MultiDiGraph with 12 nodes and 11     │
│                                       │                 │                │             │ edges                                 │
└───────────────────────────────────────┴─────────────────┴────────────────┴─────────────┴───────────────────────────────────────┘
```

Commit:
docs(running.md): add the justfile option to the docs

Anyitechs commented:
> As a new user, I want to run most of these commands using the `justfile`. Is it possible to direct users to the `justfile` features and then give them the option to explore the nitty gritty details?
> 
> One idea is to provide the justfile command up front and then give users the option to explore that command in more detail using [collapsing sections](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab)
> 
> ## Running Warnet
> `just start` - starts a minikube cluster
> 
> start details
> `just startd` - starts a minikube cluster in debug mode
> 
> startd details

Thank you for this. I've effected the changes using collapsing sections.

Commit:
docs(running.md): add port forwarding heading in docs and format bash commands properly

Commit:
docs(running.md): format bash commands properly

pinheadmz commented:
> is it still a "Docker network"

good catch @mplsgrant the title of that column should probably be "network name"

Commit:
docs(running.md): format bash commands properly on Github

Anyitechs commented:
I've addressed all the feedbacks here @pinheadmz.

Let me know if there's anything I'm missing.

mplsgrant commented on docs/running.md:
```diff
@@ -9,14 +9,224 @@ a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet
+
+</details>
+
+### Running the commands directly
```
Is it possible to merge these next few section with the section above? I am thinking about new users who come in and want to get going quickly. They will get overwhelmed by the idea of running the commands directly. 

I think advanced users will have the ability to expand the command details in each section and absorb the nitty gritty details, but having the commands listed in both the collapsed details section and again in this section will make it hard to maintain the documentation and will confuse new users.

mplsgrant commented on docs/running.md:
```diff
@@ -9,14 +9,224 @@ a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet
 
```
We need to say something about running warnet with minikube (which runs warnet locally). The Installation documentation nudges users towards minikube which means that users will expect to see how their newly-installed minikube software will be put to use.

mplsgrant replied:
Can we move this to a "Logging" section (maybe even in another file in our documentation).

> If the `$XDG_STATE_HOME` environment variable is set, the server will log to
> a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet/warnet.log`.

The reason why I think that it's important to move this information is because new users will want to hyper focus on getting their system running which means that this could distract them. Feels like premature configuration.

mplsgrant commented on docs/running.md:
```diff
@@ -9,14 +9,224 @@ a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet
 
 ## Kubernetes
```
I think we should call this "Starting Kubernetes", and we should make sure the user has run `minikube start` or whatever the equivalent is for non-minikube setups.

mplsgrant commented on docs/running.md:
```diff
@@ -9,14 +9,224 @@ a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet
 ## Kubernetes
 
-Deploy the resources in `src/templates/`, this sets up the correct permissions on the cluster (`rbac-config.yaml`) and deploys the warnet RPC server as a service + statefulset.
+Deploy the resources in `src/templates/`, this sets up a Kubernetes cluster, apply correct permissions on the cluster (`rbac-config.yaml`), and deploys the warnet RPC server as a service + statefulset.
```
I think we can get rid of this information. New users will be distracted by "src/templates", and I think advanced users will discover this information by looking at our collapsed sections.

mplsgrant commented on docs/running.md:
```diff
@@ -9,14 +9,224 @@ a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet
+Deploy the resources in `src/templates/`, this sets up a Kubernetes cluster, apply correct permissions on the cluster (`rbac-config.yaml`), and deploys the warnet RPC server as a service + statefulset.
 
-This can be done with from inside the `src/templates/` directory by running:
+### Using Justfile
```
Can we promote this to a 2-hash header? It seems like the next important step after getting minikube started.

mplsgrant commented on docs/running.md:
````diff
@@ -9,14 +9,224 @@ a file `$XDG_STATE_HOME/warnet/warnet.log`, otherwise it will use `$HOME/.warnet
+
 ```
 
+### Setup and Start RPC in Development Mode with Docker Desktop
````
~~I'm not sure this section and the next deal with Docker Desktop. That said, I have not used Docker Desktop yet. So, maybe I'm wrong about this??~~

Looks like I am wrong about this section and Docker Desktop. However, we probably want to roll up the details into the collapsed sections.

mplsgrant reviewed (changes requested):
All of my comments are geared towards the idea that a new user will come to our project and want to get warnet running on their computer right away. It might be the case that we need two pieces of documentation: a Quick Start Documentation and a Full Documentation.

Commit:
docs(running.md): clean up docs and move command details into collapsed section

Anyitechs commented:
Thank you @mplsgrant for the detailed review and feedbacks. I've addressed all the feedbacks and moved the logging details into a seperate file in the documentation ([logging.md](https://github.com/Anyitechs/warnet/blob/docs-update/docs/logging.md)).

Let me know if I missed anything.

cc: @pinheadmz

Commit:
docs(running.md): clean up docs and move command details into collapsed section

Commit:
docs(logging.md): move logging details into a new logging.md file in the documentation

mplsgrant commented on docs/running.md:
```diff
@@ -4,18 +4,223 @@ Warnet runs a server which can be used to manage multiple networks. On docker
+## Starting Kubernetes
 
-## Kubernetes
+Before proceeding, ensure that you have started your Kubernetes cluster using Minikube (`minikube start`) or any equivalent setup.
```
Quick update: as of the recent #349 we no longer need to do `minikube start`.

mplsgrant commented:
At a high level, one thing I would like to do after completing the steps in Running Warnet is I want to be able to run this: `warcli scenarios run tx_flood`. Not because tx_flood is special, but rather it let's the user know they are now set up with a warnet installation.

This way, new users can feel great about getting warnet installed using the steps from the Installation page, and then they can feel even better seeing warnet running after completing the steps in Running Warnet. I really want users to be able to step through those two pages and have something really cool to look at and play with.

willcl-ark commented:
Going to close this for now due to inactivity.

Thanks for your contribution nonetheless, please feel free to re-open in the future if you update things.
